src/text/line.rs

```diff
@@ -586,31 +555,75 @@ impl WidgetRef for Line<'_> {
     fn render_ref(&self, area: Rect, buf: &mut Buffer) {
         let area = area.intersection(buf.area);
         buf.set_style(area, self.style);
-        let width = self.width() as u16;
-        let mut x = area.left();
-        let line = if width > area.width {
-            self.truncated(area.width)
+        let line_width = self.width() as u16;
+        if line_width <= area.width {
+            let offset = match self.alignment {
+                Some(Alignment::Center) => (area.width.saturating_sub(line_width)) / 2,
+                Some(Alignment::Right) => area.width.saturating_sub(line_width),
+                Some(Alignment::Left) | None => 0,
+            };
+            let area = area.indent_x(offset);
+            render_spans(&self.spans, area, buf, 0);
         } else {
             let offset = match self.alignment {
-                Some(Alignment::Center) => (area.width.saturating_sub(width)) / 2,
-                Some(Alignment::Right) => area.width.saturating_sub(width),
+                Some(Alignment::Center) => (line_width.saturating_sub(area.width)) / 2,
+                Some(Alignment::Right) => line_width.saturating_sub(area.width),
                 Some(Alignment::Left) | None => 0,
             };
-            x = x.saturating_add(offset);
-            self.to_owned()
+            render_spans(&self.spans, area, buf, offset);
```

Comment:
When `line_width` is less than `area.width`, offset is only used to "indent" the area rect, but when `line_width` is greater than `area.width`, offset is the "span offset".

I think it may make sense to break this down into two parts? 

```rust
let line_width = self.width() as u16;
let has_enough_room = line_width <= area.width;
let offset = match (has_enough_room, alignment) {
    (true, Some(Alignment::Center)) => (area.width.saturating_sub(line_width)) / 2,
    (true, Some(Alignment::Right)) => area.width.saturating_sub(line_width),
    (false, Some(Alignment::Center)) => (line_width.saturating_sub(area.width)) / 2,
    (false, Some(Alignment::Right)) => line_width.saturating_sub(area.width),
    (_, Some(Alignment::Left) | None) => 0,
}
if has_enough_room {
    // we have enough room to render the entire line
    // shift the x coordinate of the rect and render all the spans 
    // starting from the 0th offset into the new inner area
    let inner_area = area.indent_x(offset);
    render_spans(&self.spans, inner_area, buf, 0)
} else {
    // we don't have enough room, and have to truncate the spans based on the starting offset
    render_spans(&self.spans, area, buf, offset)
}
```

Reply:
I thought about this, but I think the if else is more clear for this here? Putting both into match only creates more branch points… So I think it's both easier to understand in the if and probably more performant? But the latter is pure speculation as LLVM optimized so much stuff.

Reply:
Added comments and modified variable names to indent_width and skip_width in 1d15ef6b